### DISCUSSION TITLE
Mysql Boolean datatype column when value is False.

### DISCUSSION THREAD

**Discussion by agaheu** [Budibase/budibase#4892] [General]
Hi,

Boolean columns behaving in a way that I can't understand when using mysql datasource, details:

A boolean column ("Is Active"):
![image](https://user-images.githubusercontent.com/3132816/157985092-a2155753-1714-4946-b534-c827fa9813b9.png)

and "Formula" column which has a expression: "return $("Is Active"); "
![image](https://user-images.githubusercontent.com/3132816/157985289-9d877d00-cbaf-483c-82b3-8c202f336645.png)

For True and False values different results occour. When the boolean column is True the expression returns "1" as expected but when it's False it returns nothing:
![image](https://user-images.githubusercontent.com/3132816/157985481-2a4bc279-70b8-44bf-a9bc-8d95e49530da.png)

When I change the expression to "return typeof $("Is Active");" in order to understand the problem, the result is :
 
![image](https://user-images.githubusercontent.com/3132816/157985718-144e2f99-849c-442e-840e-7ed34b58f95d.png)

Is it some kind of bug or am i missing something?

**Comment by mjashanks** [comment]
Hey @agaheu - thanks for this - seems like a bug. Just to clarify - the value stored in your DB is definitely `false` / `0`? Or is it `null`?

I'll look into it :)

**Reply by agaheu** [reply]
It is 0
